rmgpy/rmg/modelTest.py

```diff
@@ -182,7 +190,64 @@ def testMakeNewReaction(self):
                     counter += len(rxnList)
 
         self.assertEquals(counter, 3)
+    
+    def testThermodynamicFiltering(self):
+        """
+        test that Thermodynamic pruning functionalities work
+        """
+        cerm = CoreEdgeReactionModel()
 
+        spcs = [Species().fromSMILES('[OH]'), 
+                Species().fromSMILES('C'),
+                Species().fromSMILES('[CH3]'),
+                Species().fromSMILES('[CH2]'),
+                Species().fromSMILES('O')]
+        
+        for spc in spcs:
+            cerm.makeNewSpecies(spc,label=spc.molecule[0].toSMILES())
+            spc.label = spc.molecule[0].toSMILES()
+            submit(spc)
+        
+        for spc in spcs[:3]:
+            cerm.addSpeciesToCore(spc)
+            
+        reaction = TemplateReaction(
+            reactants = [spcs[0],spcs[2]],
+            products = [spcs[-1],spcs[-2]],
+            degeneracy = 1,
+            reversible = True,
+            family = 'H_Abstraction',
+        )
+        
+        cerm.setThermodynamicFilteringParameters(Tmax=300.0, 
+                                                 toleranceThermoKeepSpeciesInEdge=0.0,
+                                                 minCoreSizeForPrune=0,
+                                                 maximumEdgeSpecies=1,
+                                                 reactionSystems=[])
+        
+        cerm.processNewReactions(newReactions=[reaction],newSpecies=[])
+        
+        edgeSpcLabels = [x.label for x in cerm.edge.species]
+        self.assertNotIn(spcs[-2].label, edgeSpcLabels)
+        
+        cerm.setThermodynamicFilteringParameters(Tmax=300.0, 
+                                                 toleranceThermoKeepSpeciesInEdge=1000.0,
+                                                 minCoreSizeForPrune=0,
+                                                 maximumEdgeSpecies=1,
+                                                 reactionSystems=[])
+        
+            
+        cerm.processNewReactions(newReactions=[reaction],newSpecies=[])
+        
+        edgeSpcLabels = [x.label for x in cerm.edge.species]
+        self.assertIn(spcs[-1].label, edgeSpcLabels)
+        self.assertIn(spcs[-2].label, edgeSpcLabels)
+        
+        cerm.thermoFilterDown(maximumEdgeSpecies=1)
+            
+        edgeSpcLabels = [x.label for x in cerm.edge.species]
+        self.assertNotIn(spcs[-2].label,edgeSpcLabels)
```

Comment:
I don't understand exactly what is being checked here and how it is different than the first one. Wasn't 'CH2' already removed during the first statement? Could you describe what is going on in the docstring for the method?